### ISSUE TITLE
Android app only receives one location update at the start

### ISSUE THREAD

**Issue by loolooii** [mauron85/cordova-plugin-background-geolocation#125]
## Your Environment

- Plugin version: 2.1.2
- Platform: Android
- Device manufacturer and model: Nexus 7 2013 (Google, Asus)
- Cordova version (`cordova -v`): 6.3.0
- Cordova platform version (`cordova platform ls`): Android 5.2.1
- Plugin configuration options: 

```
let config = {
                desiredAccuracy: 10,
                stationaryRadius: 1,
                distanceFilter: 1,
                interval: 5000,
                debug: true, //  enable this hear sounds for background-geolocation life-cycle.
                stopOnTerminate: false, // enable this to clear background location settings when the app terminates
        };
```
## Expected Behavior

I expect that I get location updates in the app (periodic) according to the interval specified. 
## Actual Behavior

I used the example in this page and adjusted configurations (I'm using Ionic 2):
http://ionicframework.com/docs/v2/native/backgroundgeolocation/
## Steps to Reproduce

I run `ionic run android --device -lc` to run on the Android device. I haven't tested on iOS because xcode tells me that the plugin has not been installed and it actually is.
## Context

Basically I want to receive location updates every 10 seconds or so, I was just changing config to test things out. 

I hope I am doing something wrong.

**Comment by loolooii** [comment]
@mauron85 could you be so kind to look into this for me? Any idea what could be wrong?

**Comment by mauron85** [comment]
add following config options:
`locationProvider`: backgroundGeolocation.provider.ANDROID_ACTIVITY_PROVIDER,
and `stopOnStillActivity`: false

**Comment by loolooii** [comment]
@mauron85  Thanks! But if you look at the example `BackgroundGeolocation` does not have property `provider`. Do you know how I can access that?

**Comment by stevenbeeckman** [comment]
Yes, same problem (also on Ionic 2 & TypeScript).

**Comment by loolooii** [comment]
I just tried `1` which is the actual value of `backgroundGeolocation.provider.ANDROID_ACTIVITY_PROVIDER` and used `stopOnStillActivity: false`, but it still gives me only 1 location update.

I do receive the notification that background tracking is enabled, so the device must be doing something.

**Comment by mauron85** [comment]
Please try latest version 2.2.0-alpha.1. And send me debug logs.

https://github.com/mauron85/cordova-plugin-background-geolocation#debugging

**Comment by loolooii** [comment]
`backgroundGeolocation` is undefined for me. I'm using the Ionic 2 example which has `BackgroundGeolocation` and that one does not have `getLogEntries()`, how else can I access this method?

**Comment by loolooii** [comment]
Btw, still the same problem on `2.2.0-alpha.1`

**Comment by mauron85** [comment]
>backgroundGeolocation is undefined for me. I'm using the Ionic 2 example which has BackgroundGeolocation and that one does not have getLogEntries(), how else can I access this method?

So how does you actually update plugin version?

**Comment by loolooii** [comment]
@mauron85 I just removed the plugin and added it again using:

`cordova plugin add cordova-plugin-mauron85-background-geolocation@2.2.0-alpha.1`

**Comment by mauron85** [comment]
Sorry, I'am not ionic expert so cannot help here.

**Comment by felipecamposclarke** [comment]
I have a similar problem in ios, only receives one location update at the start and once when the application is running in the background.

this is my code (plugin version 2.1.2):

```
backgroundGeolocation.configure(function(){
	// api call
}, function(){
	// error log
}, {
	desiredAccuracy: 10,
	stationaryRadius: 20,
	distanceFilter: 30,
	interval: 5000,
	stopOnTerminate: false,
	stopOnStillActivity: false,
	locationProvider: backgroundGeolocation.provider.ANDROID_ACTIVITY_PROVIDER,
	debug: true
});
```
This should call my endpoint every 5 seconds, or am I doing something wrong?

**Comment by mauron85** [comment]
I cannot tell you without logs. Try to use version 2.2.0-alpha.6 and send me logs https://github.com/mauron85/cordova-plugin-background-geolocation#debugging

**Comment by felipecamposclarke** [comment]
Hi @mauron85,

I'm trying to see the logs but I get a null value.

**Comment by xiaofanwu** [comment]
I am facing the same problem.If anyone solved it, please let everyone else know!!!

**Comment by vincenzoiacovone** [comment]
Same problem

**Comment by loolooii** [comment]
I eventually went for Android native, because I couldn't handle all these plugins. 

What do you guys have as `distanceFilter`? 

In native Android there is a method called `setSmallestDisplacement()` which forces location updates to come in only if the device has moved a certain number of meters. Read about it [here](https://developers.google.com/android/reference/com/google/android/gms/location/LocationRequest.html#getSmallestDisplacement())

I think `distanceFilter` is the equivalent here? Anyway, I strongly think the problem lies witch device movement. Try to set `distanceFilter` low and walk around and see if you receive any updates. Good luck.

**Comment by mauron85** [comment]
Still waiting for any debug logs.

**Comment by AsimNet** [comment]
same problem here, with version 2.3.0
I got null value!

**Comment by mauron85** [comment]
Notice: this issue has been closed because it has been reported for branch 2.x or later. It could be already fixed in v3 or is not relevant anymore. You may reopen this issue if it has been closed in error.
